src/Database/LSMTree/Monoidal.hs

```diff
@@ -306,7 +307,10 @@ readCursor ::
   => Int
   -> Cursor m k v
   -> m (V.Vector (QueryResult k v))
-readCursor = undefined
+readCursor n (Internal.MonoidalCursor c) =
+    Internal.readCursor const n c $ \k v mblob ->
+      assert (null mblob) $
```

Comment:
This should probably be an `error` instead of an `assert`, like we do in `toMonoidalLookupResult`

Reply:
Good catch!

Reply:
Not sure. It's an extra check for something that should never happen and we can handle it pretty gracefully. Similarly, we don't fail if there are mupserts when reading from a normal cursor, since the internal code just resolves it with the next entries, so it becomes indistinguishable from a single insert before being passed to the normal-specific code.